### PR TITLE
python-modules/pyahocorasick: allow support for byte strings

### PR DESCRIPTION
**Pull request by leungbk**
-------

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by armijnhemel — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
The commit message is wrong. It should start with ``python3Packages.pyahocorasick: ``.

File: pkgs/development/python-modules/pyahocorasick/default.nix [hunk 1]
```diff
@@ -4,6 +4,7 @@
 , fetchpatch
 , pytestCheckHook
 , pythonOlder
+, buildAsUnicode ? true
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyahocorasick/default.nix:7]:
```suggestion
```
That is mostly forbidden in pythonPackages due to pip not handling multiple variants of the same package.

File: pkgs/development/python-modules/pyahocorasick/default.nix [hunk 2]
```diff
@@ -20,6 +21,10 @@ buildPythonPackage rec {
     hash = "sha256-APpL99kOwzIQjePvRDeJ0FDm1kjBi6083JMKuBqtaRk=";
   };
 
+  preBuild = lib.optionalString (!buildAsUnicode) ''
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyahocorasick/default.nix:24]:
```suggestion
  preBuild = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by leungbk** [comment]
> That is mostly forbidden in pythonPackages due to pip not handling multiple variants of the same package.

@SuperSandro2000 I wasn't aware that pip wouldn't handle this. Would [this](https://github.com/armijnhemel/binaryanalysis-ng/pull/330/files) be a better way of doing it (having the end user manually set `preBuild` as opposed to defining an option in nixpkgs)?

If so, since the input flag I proposed, when `true`, actually disables my proposed `preBuild` text (as opposed to enabling it, as your recommended fix seems to suggest), I suppose the best thing to do is to set the flag downstream and close this PR.

**Pull request closed without merging**